evalml/tests/automl_tests/test_automl.py

```diff
@@ -2701,3 +2780,17 @@ def test_automl_issues_beta_warning_for_time_series(problem_type, X_y_binary):
         assert len(warn) == 1
         message = "Time series support in evalml is still in beta, which means we are still actively building its core features"
         assert str(warn[0].message).startswith(message)
+
+
+def test_automl_validates_data_passed_in_to_allowed_pipelines(X_y_binary, dummy_binary_pipeline_class):
```

Comment:
I love your tests and I love that your exceptions are screaming at the user.  Absolutely love that.  If it were me, all the exceptions would be in caps ala 1994 AOL messages from your parents.